gps_common/src/utm_odometry_node.cpp

```diff
@@ -3,245 +3,170 @@
+  world_odom_pub.publish(odom_world);
+}
 
 int main (int argc, char **argv) {
   ros::init(argc, argv, "utm_odometry_node");
   ros::NodeHandle node;
   ros::NodeHandle priv_node("~");
-  bool use_ins = true;
-  odom_init = false; //initialization flag for odom.
-  publish_odom_tf = false;
-  broadcast_world_tf = false;
   priv_node.param<std::string>("frame_id", frame_id, "");
   priv_node.param<std::string>("child_frame_id", child_frame_id, "");
-  priv_node.param<double>("rot_covariance", rot_cov, 99999.0);
-
-  priv_node.param<std::string>("world_frame_id", world_frame_id, "");
-  priv_node.param<std::string>("world_child_frame_id", world_child_frame_id, "");
-
-
-  if(!priv_node.getParam("use_ins", use_ins))
-	  ROS_WARN_STREAM("No use_ins param provided, using default: "<<use_ins);
-
-  if(!priv_node.getParam("publish_odom_tf", publish_odom_tf))
-	  ROS_WARN_STREAM("No publish_odom_tf param provided, using default: "<<publish_odom_tf);
-
-  if(!priv_node.getParam("broadcast_world_tf",broadcast_world_tf))
-	  ROS_WARN_STREAM("No world_tf param provided, using default: "<<broadcast_world_tf);
-
-
-  if(use_ins)
-  {
-	  odom_pub = node.advertise<nav_msgs::Odometry>("/odom", 10);
-	  world_odom_pub = node.advertise<nav_msgs::Odometry>("/odom_world", 10);
-	  gps_source_sub = node.subscribe("/inspva", 10, insCallBack);
-  }
-  else
-  {
-	  odom_pub = node.advertise<nav_msgs::Odometry>("odom", 10);
-	  gps_source_sub = node.subscribe("fix", 10, callback);
+  priv_node.param<std::string>("world_frame_id", world_frame_id, "world");
+  priv_node.param<bool>("publish_odom_tf", publish_odom_tf, false);
+  priv_node.param<bool>("ignore_z", ignore_z, false);
+
+  // more than parameters, a service call would be better
+  // so that we could change the origin at runtime
+  geometry_msgs::PoseStamped origin;
+  priv_node.param<bool>("use_fixed_origin", use_fixed_origin, false);
+  priv_node.param<double>("origin_x", origin.pose.position.x, 0);
+  priv_node.param<double>("origin_y", origin.pose.position.y, 0);
+  priv_node.param<double>("origin_z", origin.pose.position.z, 0);
+  // using euler angles might be easier
+  priv_node.param<double>("origin_ox", origin.pose.orientation.x, 0);
+  priv_node.param<double>("origin_oy", origin.pose.orientation.y, 0);
+  priv_node.param<double>("origin_oz", origin.pose.orientation.z, 0);
+  priv_node.param<double>("origin_ow", origin.pose.orientation.w, 1);
+
+  if (!use_fixed_origin) {
+    auto ins = ros::topic::waitForMessage<novatel_gps_msgs::Inspva>("/inspva", node);
+    double northing, easting;
+    std::string zone;
+    LLtoUTM(ins->latitude, ins->longitude, northing, easting, zone);
+    origin.pose.position.x = easting;
+    origin.pose.position.y = northing;
+    origin.pose.position.z = ins->height;
+    origin.pose.orientation = tf::createQuaternionMsgFromRollPitchYaw(
+      ins->roll * degrees_to_radians,
+      ins->pitch * degrees_to_radians,
+       (90.0 - ins->azimuth) * degrees_to_radians);
```

Comment:
maybe there is a miss alignment here? I don't know if it's a common practice in c++

Reply:
bad indentation. 

- [ ] fix indentation.